keys/cmd/create.go

```diff
@@ -0,0 +1,163 @@
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client/input"
+	"github.com/cosmos/go-bip39"
+	"github.com/spf13/cobra"
+	"github.com/syndtr/goleveldb/leveldb"
+	cryptoPocket "keys/crypto"
+)
+
+const (
+	flagRecover = "recover"
+)
+
+// createCmd represents the create command
+var createCmd = &cobra.Command{
+	Use:   "create <name>",
+	Short: "Creating an encrypted private key and save to <name> file as the key pair identifier",
+	Long: `Derive a new private key and encrypt to disk.
+
+Allow users to use BIP39 mnemonic and to secure the mnemonic. Take key ID <name> from user and store key under the <name>.
+`,
+	Args: cobra.ExactArgs(1),
+	RunE: runAddCmd,
+}
+
+/*
+key type
+	- name: the unique ID for the key
+	- publickey: the public key
+	- privatekey: the private key
+	- Address: the address related to the private key
+	- mnemonic: mnemonic used to generated key, empty if not saved
+*/
+type key struct {
+	Name       string                  `json:"name"`
+	PublicKey  cryptoPocket.PublicKey  `json:"publickey"`
+	PrivateKey cryptoPocket.PrivateKey `json:"privatekey"`
+	Address    cryptoPocket.Address    `json:"address"`
+	Mnemonic   string                  `json:"mnemonic"`
+}
+
+// Future updates
+// - determine a safer keystore location (team discuss)
+// - confirmation from user for overriding existing key
+// - implement key phrase intput from user secure keys
+func runAddCmd(cmd *cobra.Command, args []string) error {
+	var err error
+	var inBuf = bufio.NewReader(cmd.InOrStdin())
+
+	name := args[0]
+
+	//////////////////////////
+	//  Mnemonic Generation //
+	//////////////////////////
+
+	// Get bip39 mnemonic
+	var mnemonic string
+	var bip39Passphrase string = ""
+
+	// User can recover private key from mnemonic
+	recover, err := cmd.Flags().GetBool(flagRecover)
+	if err != nil {
+		return err
+	}
+
+	if recover {
+		mnemonic, err = input.GetString("Enter your bip39 mnemonic", inBuf)
+		if err != nil {
+			return err
+		}
+
+		if !bip39.IsMnemonicValid(mnemonic) {
+			return errors.New("invalid mnemonic")
+		}
+	} else {
+		// read entropy seed straight from tmcrypto.Rand and convert to mnemonic
+		entropySeed, err := bip39.NewEntropy(mnemonicEntropySize)
+		if err != nil {
+			return err
+		}
+
+		mnemonic, err = bip39.NewMnemonic(entropySeed)
+		if err != nil {
+			return err
+		}
+	}
+
+	/////////////////////
+	// Keys Generation //
+	/////////////////////
+
+	kb, err := leveldb.OpenFile("./.keybase/poktKeys.db", nil)
+	if err != nil {
+		return err
+	}
+
+	// Creating a private key with ED25519 and mnemonic seed phrases
+	privateKey, err := cryptoPocket.NewPrivateKeyFromSeed([]byte(mnemonic + bip39Passphrase))
+	if err != nil {
+		return err
+	}
+
+	// Creating a public key
+	publicKey := privateKey.PublicKey()
+
+	// Creating an address
+	address := privateKey.Address()
+
+	// JSON encoding
+	var keystore = key{name, publicKey, privateKey, address, mnemonic}
```

Comment:
It appears as if we are storing `clear text` `non obfuscated` private keys in a leveldb. This is not up to the V0 implementation's security requirements. 

[V0 Keybase Implementation](https://github.com/pokt-network/pocket-core/blob/40c52a1cf3d4ccc6ed161b2ac1853a003c32324f/crypto/keys/keybase.go#L238)